### PR TITLE
fix(region, host): misc fix host health checker

### PR DESCRIPTION
**Pull request by wanyaoqi**
hostagent remove option enable host health, now default is enabled,
its rely on etcd endpoint register.
add option auto migrate on host shutdown.
fix hostman check network is available.

Signed-off-by: wanyaoqi <wanyaoqi@yunion.cn>

**What this PR does / why we need it**:

**Does this PR need to be backport to the previous release branch?**:

### CODE REVIEW

File: pkg/compute/models/hosts.go [hunk 1]
```diff
@@ -3984,12 +3985,42 @@ func (self *SHost) PerformOnline(ctx context.Context, userCred mcclient.TokenCre
 func (self *SHost) PerformAutoMigrateOnHostDown(
 	ctx context.Context, userCred mcclient.TokenCredential, query jsonutils.JSONObject, data jsonutils.JSONObject,
 ) (jsonutils.JSONObject, error) {
-	val, _ := data.GetString("auto_migrate_on_host_down")
+	var meta = make(map[string]interface{})
```
Comment (ioito) [inline pkg/compute/models/hosts.go:3988]: 把入参改成struct，别再用JSONObject了

File: pkg/compute/models/hosts.go [hunk 2]
```diff
@@ -3984,12 +3985,42 @@ func (self *SHost) PerformOnline(ctx context.Context, userCred mcclient.TokenCre
...
+func (self *SHost) PerformAutoMigrateOnHostShutdown(
+	ctx context.Context, userCred mcclient.TokenCredential, query jsonutils.JSONObject, data jsonutils.JSONObject,
+) (jsonutils.JSONObject, error) {
+	val, _ := data.GetString("auto_migrate_on_host_shutdown")
 
 	var meta map[string]interface{}
 	if val == "enable" {
 		meta = map[string]interface{}{
-			"__auto_migrate_on_host_down": "enable",
+			api.HOSTMETA_AUTO_MIGRATE_ON_HOST_DOWN: "enable",
```
Comment (ioito) [inline pkg/compute/models/hosts.go:4023]: 入参->struct

File: pkg/compute/models/hosts.go (hunk 1 above)
Comment (wanyaoqi) [reply pkg/compute/models/hosts.go:3990]: fixed

File: pkg/compute/models/hosts.go (hunk 2 above)
Comment (wanyaoqi) [reply pkg/compute/models/hosts.go]: 这里是设置 metadata

**Pull request merged**